service/provider/src/main/java/gov/va/vro/service/event/AuditEventProcessor.java

```diff
@@ -0,0 +1,60 @@
+package gov.va.vro.service.event;
+
+import gov.va.vro.camel.FunctionProcessor;
+import gov.va.vro.model.event.AuditEvent;
+import gov.va.vro.model.event.Auditable;
+import gov.va.vro.service.spi.audit.EventService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class AuditEventProcessor {
+
+  private final EventService eventService;
+
+  public Object logEvent(Object o, String routeId, String message) {
+    if (!(o instanceof Auditable auditableObject)) {
+      // object cannot be audited
```

Comment:
Not sure why we are expecting not Auditable object in the arguments and using an object. I would say that would be a software error. I would at least add a log.error or log.info here based on that what the thinking is.

Reply:
There is a general exception handler that catches any exception, but we don't want to audit all exceptions, only the ones for objects that we are auditing. Therefore the check and conversion is necessary. It is not really an error if the object is not auditable, because we just skip it. I can add a log.info to make this clear in the next ticket.